src/gromacs/nblib/tests/interactions.cpp

```diff
@@ -0,0 +1,133 @@
+ *
+ * Copyright (c) 2020, by the GROMACS development team, led by
+ * Mark Abraham, David van der Spoel, Berk Hess, and Erik Lindahl,
+ * and including many others, as listed in the AUTHORS file in the
+ * top-level source directory and at http://www.gromacs.org.
+ *
+ * GROMACS is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ *
+ * GROMACS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with GROMACS; if not, see
+ * http://www.gnu.org/licenses, or write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA.
+ *
+ * If you want to redistribute modifications to GROMACS, please
+ * consider that scientific software is very special. Version
+ * control is crucial - bugs must be traceable. We will be happy to
+ * consider code for inclusion in the official distribution, but
+ * derived work must not be called official GROMACS. Details are found
+ * in the README & COPYING files - if they are missing, get the
+ * official version at http://www.gromacs.org.
+ *
+ * To help us fund GROMACS development, we humbly ask that you cite
+ * the research papers on the package. Check out http://www.gromacs.org.
+ */
+/*! \internal \file
+ * \brief
+ * This implements molecule setup tests
+ *
+ * \author Victor Holanda <victor.holanda@cscs.ch>
+ * \author Joe Jordan <ejjordan@kth.se>
+ * \author Prashanth Kanduri <kanduri@cscs.ch>
+ * \author Sebastian Keller <keller@cscs.ch>
+ * \author Artem Zhmurov <zhmurov@gmail.com>
+ */
+#include "gmxpre.h"
+
+#include <iostream>
+
+#include "gromacs/nblib/interactions.h"
+#include "gromacs/nblib/particletype.h"
+
+#include "testutils/testasserts.h"
+#include "testsystems.h"
+
+namespace nblib
+{
+namespace test
+{
+namespace
+{
+
+TEST(NBlibTest, NonBondedForceParamsCorrect)
+{
+    ParticleType atom1(ParticleName("a1"), Mass(1));
+    ParticleType atom2(ParticleName("a2"), Mass(1));
+    ParticleType atom3(ParticleName("a3"), Mass(1));
+
+    ParticleTypesInteractions interactions;
+
+    real c6_1  = 1.6;
+    real c12_1 = 1.12;
+    real c6_2  = 2.6;
+    real c12_2 = 2.12;
+    real c6_3  = 3.6;
+    real c12_3 = 3.12;
+
+    real c6comb  = 40;
+    real c12comb = 41;
+
+    interactions.add(atom1.name(), c6_1, c12_1);
+    interactions.add(atom2.name(), c6_2, c12_2);
+    interactions.add(atom3.name(), c6_3, c12_3);
+    interactions.add(atom2.name(), atom3.name(), c6comb, c12comb);
+
+    auto nbfp = interactions.generateTable();
+
+    //! self interaction for c6
```

Comment:
This code is complex enough that it suggest that we should add a lookup functionality to ParticleTypeInteractions.

Reply:
hmm, there's no place in the code where it would be needed atm. If we did have such an interface in the ParticleTypeInteractions class, it would be a duplicate for accessing the map, so I don't see how it would simplify the test either.

Reply:
I just think it would be nice to have a way to not have to type `std::get<0>(nbfp[std::make_tuple(atom1.name(), atom1.name())]` but instead be able to call e.g. `getInteraction(firstAtom, secondAtom, C6)`.

Reply:
Ok, this is indeed nicer.